resolwe_bio/processes/support_processors/report_variants.py

```diff
@@ -0,0 +1,502 @@
+            label="ANN fields to use",
+            description="Only use specific fields from the SnpEff ANN "
+            "field. All available fields: Allele | Annotation | Annotation_Impact "
+            "| Gene_Name | Gene_ID | Feature_Type | Feature_ID | Transcript_BioType "
+            "| Rank | HGVS.c | HGVS.p | cDNA.pos / cDNA.length | CDS.pos / CDS.length "
+            "| AA.pos / AA.length | Distance | ERRORS / WARNINGS / INFO' ."
+            "Fields are seperated by '|'. For more information, follow this [link]"
+            "(https://pcingola.github.io/SnpEff/se_inputoutput/#ann-field-vcf-output-files).",
+            default=[
+                "Allele",
+                "Annotation",
+                "Annotation_Impact",
+                "Gene_Name",
+                "Feature_ID",
+                "HGVS.p",
+            ],
+        )
+        reference = DataField(
+            data_type="variantstable",
+            label="Table of known annotated variants",
+            description="Table of dbSNP variants annotated with SnpEff.",
+        )
+        bam = DataField(
+            data_type="alignment:bam",
+            label="Bam file used for coverage calculation",
+            description="Output from BQSR or SplitNCigarReads should be used.",
+        )
+
+        class Advanced:
+            """Advanced options."""
+
+            split_alleles = BooleanField(
+                label="Split multi-allelic records into multiple lines",
+                description="By default, a variant record with multiple "
+                "ALT alleles will be summarized in one line, with per "
+                "alt-allele fields (e.g. allele depth) separated by commas."
+                "This may cause difficulty when the table is loaded by "
+                "an R script, for example. Use this flag to write multi-allelic "
+                "records on separate lines of output.",
+                default=True,
+            )
+            gf_fields = ListField(
+                StringField(),
+                label="Include FORMAT/sample-level fields",
+                default=[
+                    "GT",
+                ],
+            )
+
+        advanced = GroupField(Advanced, label="Advanced options")
+
+    class Output:
+        """Output fields to ReportVariants."""
+
+        tsv = FileField(
+            label="Mutations table",
+        )
+        genes = ListField(StringField(), label="Input genes")
+        species = StringField(label="Species")
+        build = StringField(label="Build")
+
+    def run(self, inputs, outputs):
+        """Run analysis."""
+
+        if not all(
+            field in inputs.vcf_fields for field in ["CHROM", "POS", "ID", "REF", "ANN"]
+        ):
+            self.error(
+                "Input VCF fields do not contain all required values. "
+                "Required fields are CHROM, POS, ID, REF and ANN."
+            )
+
+        if inputs.variants.entity_name != inputs.bam.entity_name:
```

Comment:
A safer way would be to check for sample id or slug, as there might be duplicated sample names in the set of possible input files.